x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/query/VectorDVLeafFieldData.java

```diff
@@ -64,30 +58,14 @@ public SortedBinaryDocValues getBytesValues() {
             if (indexed) {
                 VectorValues values = reader.getVectorValues(field);
                 if (values == null || values == VectorValues.EMPTY) {
-                    return new DelegateDocValuesField(DenseVectorScriptDocValues.empty(new DenseVectorSupplier<float[]>() {
-                        @Override
-                        public float[] getInternal() {
-                            throw new IllegalArgumentException(MISSING_VECTOR_FIELD_MESSAGE);
-                        }
-
-                        @Override
-                        public void setNextDocId(int docId) throws IOException {
-                            // do nothing
-                        }
-
-                        @Override
-                        public int size() {
-                            return 0;
-                        }
-                    }, dims), name);
+                    // There's no way for KnnDenseVectorDocValuesField to reliably differentiate between VectorValues.EMPTY and
+                    // values that can be iterated through. Since VectorValues.EMPTY throws on docID(), pass a null instead.
+                    values = null;
```

Comment:
I find this behavior to be a bit strange in that `VectorValues.EMPTY` will throw an exception, but `null` does not.

Reply:
`VectorValues.EMPTY` is part of the Lucene API, so it cannot be easily changed as part of this PR